src/sentry/integrations/slack/actions/notification.py

```diff
@@ -117,12 +126,72 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
                     "attachments": json.dumps(attachments),
                 }
 
+            rule = rules[0] if rules else None
+            rule_to_use = self.rule if self.rule else rule
+            rule_id = rule_to_use.id if rule_to_use else None
+            rule_action_uuid = self.data.get("action_uuid", None)
+            if not rule_action_uuid:
+                # We are logging because this should never happen, all actions should have an uuid
+                # We can monitor for this, and create an alert if this ever appears
+                _default_logger.info(
+                    "No action uuid",
+                    extra={
+                        "rule_id": rule_id,
+                        "notification_uuid": notification_uuid,
+                    },
+                )
+
+            new_notification_message_object = NewIssueAlertNotificationMessage(
+                rule_fire_history_id=self.rule_fire_history.id if self.rule_fire_history else None,
+                rule_action_uuid=rule_action_uuid,
+            )
```

Comment:
if `rule_fire_history` is `None` we will always error here and not send a message regardless if the org is under the feature flag. is that what we want?

Reply:
we won't error here, but we will throw a NotificationMessageValidationError exception later in this function that will then get logged. I think it's fine if that happens, because that means another path is calling this flow that probably shouldn't be